app/src/main/java/moe/apex/rule34/navigation/Destinations.kt

```diff
@@ -6,12 +6,24 @@ import kotlinx.serialization.Serializable
 import kotlin.reflect.KClass
 
 
+@Serializable
+data class ImageView(
+    val source: String,
+    val id: String
+)
+
+@Serializable
+data class DeepLinkImageView(
```

Comment:
For this it might be better to just use the `ImageView` destination that you also created. `ImageSource#loadImageFromUri` already has logic to extract the source and post ID from a link so it might be worth moving those into their own functions and passing the results when we navigate to `ImageView`.